File: backend/src/main/java/com/qa/service/AnswerService.java
package com.qa.service;

import com.qa.common.PageResult;
import com.qa.dto.AnswerDTO;
import com.qa.entity.Answer;

/**
 * 回答业务服务
 */
public interface AnswerService {
    
    /**
     * 发布回答
     */
    Answer publishAnswer(Long questionId, Long authorId, String content);
    
    /**
     * 获取回答列表
     */
    PageResult<AnswerDTO> getAnswerList(Long questionId, Integer pageNum, Integer pageSize, Long userId);
    
    /**
     * 更新回答
     */
    Answer updateAnswer(Long answerId, String content, Long userId);
    
    /**
     * 删除回答
     */
    void deleteAnswer(Long answerId, Long userId);
    
    /**
     * 获取用户的回答列表
     */
    PageResult<AnswerDTO> getUserAnswers(Long authorId, Integer pageNum, Integer pageSize, Long userId);
}

File: frontend/src/services/answerService.js
import api from '@/utils/api'

const answerService = {
  publishAnswer(questionId, content) {
    return api.post('/answers', null, {
      params: { questionId, content }
    })
  },

  getAnswerList(questionId, pageNum = 1, pageSize = 20) {
    return api.get('/answers', {
      params: { questionId, pageNum, pageSize }
    })
  },

  getUserAnswers(userId, pageNum = 1, pageSize = 20) {
    return api.get(`/answers/user/${userId}`, {
      params: { pageNum, pageSize }
    })
  },

  updateAnswer(id, content) {
    return api.put(`/answers/${id}`, null, {
      params: { content }
    })
  },

  deleteAnswer(id) {
    return api.delete(`/answers/${id}`)
  }
}

export default answerService
